flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/table/MySqlReadableMetadata.java

```diff
@@ -80,6 +80,28 @@ public Object read(SourceRecord record) {
                     return TimestampData.fromEpochMillis(
                             (Long) sourceStruct.get(AbstractSourceInfo.TIMESTAMP_KEY));
                 }
+            }),
+
+    /** The update-before data for UPDATE record. */
+    OLD(
+            "update_before",
+            DataTypes.ARRAY(
+                            DataTypes.MAP(
+                                            DataTypes.STRING().nullable(),
+                                            DataTypes.STRING().nullable())
+                                    .nullable())
+                    .nullable(),
+            new MetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Object read(SourceRecord record) {
+                    final Envelope.Operation op = Envelope.operationFor(record);
+                    if (op != Envelope.Operation.UPDATE) {
+                        return null;
+                    }
+                    return record;
```

Comment:
The `record` is the entire record instead of the `update_before` fields IIUC.
Could you also at least add a test as well as the document for the change?

Reply:
Sure, sorry about the late reply, I will update this pr asap